src/pudl/transform/ferc1.py

```diff
@@ -781,6 +785,38 @@ def check_table_calcuations(
             f"Calculations in {table_name} are off by {off_ratio}. Expected tolerance "
             f"of {params.calculation_tolerance}."
         )
+
+    # Check that sub-total calculations sum to total.
+    if params.subtotal_column is not None:
+        sub_group_col = params.subtotal_column
+        pks_wo_subgroup = [col for col in pks if col != sub_group_col]
+        calculated_df["sub_total_sum"] = (
+            calculated_df.pipe(lambda df: df[df[sub_group_col] != "total"])
+            .groupby(pks_wo_subgroup)["calculated_amount"]
+            .transform("sum")
+        )
+        calculated_df["sub_total_sum"] = calculated_df["sub_total_sum"].fillna(
+            calculated_df.calculated_amount
+        )
+        sub_total_errors = (
+            calculated_df.groupby(pks_wo_subgroup)
+            .filter(lambda x: x["sub_total_sum"].nunique() > 1)
+            .groupby(pks_wo_subgroup)
+        )
```

Comment:
i don't doubt that this is doing the thing but this feels opaque to me. can you explain it? i can guess that if a group has more than one value than its a baddie. But why not just find the instances where `calculated_df.sub_total_sum.notnull() & (calculated_df.sub_total_sum != calculated_df.calculated_amount)`? i think this is just something i am not understanding.

Reply:
Basically I fill every value in this column with the sum of the sub-components (i.e. the total as calculated from the sub-components), except for the "total" row which is filled in directly with the calculated total value. If there is more than one kind of value in the group, it means that the total from the sub-components and the calculated total aren't equivalent. I group and filter like this because I'm trying to compare values both across rows and across groups. It's a bit confusing, I can document it a bit better! But directly comparing the `sub_total_sum` to `calculated_amount` within the same row won't provide the needed information.